controller-server/src/main/java/com/yahoo/vespa/hosted/controller/tenant/TenantInfo.java

```diff
@@ -0,0 +1,106 @@
+package com.yahoo.vespa.hosted.controller.tenant;
+
+import java.util.Objects;
+
+/**
+ * Tenant information beyond technical tenant id and user authorizations.
+ *
+ * This info is used to capture generic support information and invoiced billing information.
+ *
+ * All fields are non null but strings can be empty
+ */
+public class TenantInfo {
+    private final String name;
+    private final String email;
+    private final String website;
+    private final String contactName;
+    private final String contactEmail;
+    private final String invoiceEmail;
+    private final TenantInfoAddress address;
+    private final TenantInfoBillingContact billingContact;
+
+    TenantInfo(String name, String email, String website, String contactName, String contactEmail,
+               String invoiceEmail, TenantInfoAddress address, TenantInfoBillingContact billingContact) {
+        this.name = Objects.requireNonNull(name);
+        this.email = Objects.requireNonNull(email);
+        this.website = Objects.requireNonNull(website);
+        this.contactName = Objects.requireNonNull(contactName);
+        this.contactEmail = Objects.requireNonNull(contactEmail);
+        this.invoiceEmail = Objects.requireNonNull(invoiceEmail);
+        this.address = Objects.requireNonNull(address);
+        this.billingContact = Objects.requireNonNull(billingContact);
+    }
+
+    public static TenantInfo EmptyInfo = new TenantInfo("","","", "", "", "",
```

Comment:
Should be named simply `EMPTY`, same goes for the other classes.

Reply:
I wasn't aware of that. Is there a naming convention document that I have missed?

Reply:
In general the Java convention is `UPPERCASE` for constants. In this module we typically name such constants either `EMPTY/NONE` or `empty/none`.

Reply:
Also, this should be `final`.

Reply:
fixed